dev/NumberBox/NumberBox.cpp

```diff
@@ -0,0 +1,387 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "NumberBox.h"
+#include "NumberBoxAutomationPeer.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+#include "Utils.h"
+
+NumberBox::NumberBox()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_NumberBox);
+
+    // Default values for the number formatter
+    auto formatter = winrt::DecimalFormatter();
+    formatter.IntegerDigits(1);
+    formatter.FractionDigits(0);
+    NumberFormatter(formatter);
+
+    PointerWheelChanged({ this, &NumberBox::OnScroll });
```

Comment:
does this need a revoker?

Reply:
Attaching to events on "self" is fine to do as the event's lifetime is the same as the target (self) by definition. No need for revoker in this scenario. It would be cool to capture this as a guideline/protip somewhere if we could.

---
In reply to: [340878649](https://github.com/microsoft/microsoft-ui-xaml/pull/1507#discussion_r340878649) [](ancestors = 340878649)